Formula/unzip.rb

```diff
@@ -15,31 +15,41 @@ class Unzip < Formula
 
   keg_only :provided_by_osx
 
-  # Upstream is unmaintained so we use the Ubuntu unzip-6.0-20ubuntu1 patchset:
-  # http://changelogs.ubuntu.com/changelogs/pool/main/u/unzip/unzip_6.0-20ubuntu1/changelog
+  # Upstream is unmaintained so we use the Debian patchset:
+  # https://packages.debian.org/sid/unzip
+  patch do
+    url "https://mirrors.ocf.berkeley.edu/debian/pool/main/u/unzip/unzip_6.0-21.debian.tar.xz"
+    mirror "https://mirrorservice.org/sites/ftp.debian.org/debian/pool/main/u/unzip/unzip_6.0-21.debian.tar.xz"
+    sha256 "8accd9d214630a366476437a3ec1842f2e057fdce16042a7b19ee569c33490a3"
+    apply %w[
+      patches/01-manpages-in-section-1-not-in-section-1l.patch
+      patches/02-this-is-debian-unzip.patch
+      patches/03-include-unistd-for-kfreebsd.patch
+      patches/04-handle-pkware-verification-bit.patch
+      patches/05-fix-uid-gid-handling.patch
+      patches/06-initialize-the-symlink-flag.patch
+      patches/07-increase-size-of-cfactorstr.patch
+      patches/08-allow-greater-hostver-values.patch
+      patches/09-cve-2014-8139-crc-overflow.patch
+      patches/10-cve-2014-8140-test-compr-eb.patch
+      patches/11-cve-2014-8141-getzip64data.patch
+      patches/12-cve-2014-9636-test-compr-eb.patch
+      patches/13-remove-build-date.patch
+      patches/14-cve-2015-7696.patch
+      patches/15-cve-2015-7697.patch
+      patches/16-fix-integer-underflow-csiz-decrypted.patch
+      patches/17-restore-unix-timestamps-accurately.patch
+      patches/18-cve-2014-9913-unzip-buffer-overflow.patch
+      patches/19-cve-2016-9844-zipinfo-buffer-overflow.patch
+    ]
+  end
+
+  # Debian doesn't carry the nzip60-alt-iconv-utf8 patch for some reason.
   patch do
     url "https://launchpad.net/ubuntu/+archive/primary/+files/unzip_6.0-20ubuntu1.debian.tar.xz"
     mirror "https://www.mirrorservice.org/sites/archive.ubuntu.com/ubuntu/pool/main/u/unzip/unzip_6.0-20ubuntu1.debian.tar.xz"
     sha256 "0ddf122ef15b739e3ea06db4b9e80f40759dce23a2c886678881453a43bd0842"
-    apply %w[
-      patches/01-manpages-in-section-1-not-in-section-1l
-      patches/02-branding-patch-this-is-debian-unzip
-      patches/03-include-unistd-for-kfreebsd
-      patches/04-handle-pkware-verification-bit
-      patches/05-fix-uid-gid-handling
-      patches/06-initialize-the-symlink-flag
-      patches/07-increase-size-of-cfactorstr
-      patches/08-allow-greater-hostver-values
-      patches/09-cve-2014-8139-crc-overflow
-      patches/10-cve-2014-8140-test-compr-eb
-      patches/11-cve-2014-8141-getzip64data
-      patches/12-cve-2014-9636-test-compr-eb
-      patches/13-remove-build-date
-      patches/14-cve-2015-7696
-      patches/15-cve-2015-7697
-      patches/16-fix-integer-underflow-csiz-decrypted
-      patches/20-unzip60-alt-iconv-utf8
-    ]
+    apply "patches/20-unzip60-alt-iconv-utf8"
```

Comment:
Actually `to fix buffer overflow in charset_to_intern()` makes it sound fairly mandatory, since that's a vuln 😓.